content/webapp/services/prismic/fetch/articles.ts

```diff
@@ -1,61 +1,35 @@
 import { Query } from '@prismicio/types';
-import { GetServerSidePropsPrismicClient } from '.';
+import { fetcher, GetServerSidePropsPrismicClient, GetByTypeParams } from '.';
 import { ArticlePrismicDocument, articlesFetchLinks } from '../types/articles';
 import { graphQuery } from '@weco/common/services/prismic/articles';
 
-const fetchLinks = articlesFetchLinks;
-
-// For other types we use fetcher.getById (see e.g. books.ts).
-//
-// That method looks for matching documents with a single type (e.g. document.type === 'books');
-// because an article can have two types, we can't use that here.
-export async function fetchArticle(
-  { client }: GetServerSidePropsPrismicClient,
-  id: string
-): Promise<ArticlePrismicDocument | undefined> {
-  try {
-    const document = await client.getByID<ArticlePrismicDocument>(id, {
-      fetchLinks,
-    });
-
-    if (document.type === 'articles' || document.type === 'webcomics') {
-      return document;
-    }
-  } catch {}
-}
-
-type Params = Parameters<
-  GetServerSidePropsPrismicClient['client']['getByType']
->[1];
-export async function fetchArticles(
-  { client }: GetServerSidePropsPrismicClient,
-  params: Params = {}
-): Promise<Query<ArticlePrismicDocument>> {
-  const orderings = ['document.first_publication_date desc'];
-  
-  /**
-   * articles and webcomics share the same functionality as we
-   * can't change the types of documents in Prismic.
-   *
-   * We thus have to fetch both here
-   * {@link} https://community.prismic.io/t/import-export-change-type-of-imported-document/7814
-   */
-  const articleAndWebcomicPredicate =
-    '[any(document.type, ["articles", "webcomics"])]';
-
-  const document = await client.get<ArticlePrismicDocument>({
-    fetchLinks,
-    orderings,
-    graphQuery,
+/**
+  * articles and webcomics share the same functionality as we
+  * can't change the types of documents in Prismic.
+  *
+  * We thus have to fetch both here
+  * {@link} https://community.prismic.io/t/import-export-change-type-of-imported-document/7814
+  */
+const articlesFetcher = fetcher<ArticlePrismicDocument>(['articles', 'webcomics'], articlesFetchLinks);
+
+export const fetchArticle = articlesFetcher.getById;
+
+export const fetchArticles = (
+  client: GetServerSidePropsPrismicClient,
+  params: GetByTypeParams
+): ReturnType<typeof articlesFetcher.getByType> => {
+  return articlesFetcher.getByType(client, {
     ...params,
-    predicates: [articleAndWebcomicPredicate, ...(params.predicates ?? [])],
+    orderings: [
+      { field: 'my.books.datePublished', direction: 'desc' },
+      { field: 'document.first_publication_date desc' },
+    ],
```

Comment:
The last time we fiddled this, we ended up with an `/articles` page full of comics and no stories. I've checked locally – it's still showing the right stuff!

Also, this means we'll get the `delistPredicate` as part of the standard fetcher – which previously we omitted here, so delisted articles could theoretically have appeared.